docs/v1-proposed-api.md

```diff
@@ -0,0 +1,254 @@
+  - generate: true
+    name: primary-cert-bundle
+    secretRef:
+      name: primary-cert-bundle-secret
+  clusterName: ci-cluster-v4-1
+  compute:
+  - name: worker
+    platform:
+      aws:
+        rootVolume:
+          iops: 100
+          size: 300
+          type: gp2
+        type: m5.xlarge
+        zones:
+        - us-east-1a
+    replicas: 4
+  controlPlane:
+    name: master
+    platform:
+      aws:
+        rootVolume:
+          iops: 100
+          size: 32
+          type: gp2
+        type: m5.xlarge
+        zones:
+        - us-east-1a
+    replicas: 3
+  controlPlaneConfig:
+    servingCertificates:
+      default: primary-cert-bundle
+  imageSet:
+    name: openshift-v4.1.18
+  images: {}
+  ingress:
+  - domain: apps.ci-cluster-v4-1.r4f1.s1.devshift.org
+    name: default
+    servingCertificate: primary-cert-bundle
+  installed: true
+  manageDNS: true
+  networking:
+    clusterNetworks:
+    - cidr: 10.128.0.0/14
+      hostSubnetLength: 23
+    machineCIDR: 10.0.0.0/16
+    serviceCIDR: 172.30.0.0/16
+    type: OpenShiftSDN
+  platform:
+    aws:
+      region: us-east-1
+  platformSecrets:
+    aws:
+      credentials:
+        name: aws
+  pullSecret:
+    name: pull
+  sshKey:
+    name: ssh
+status:
+  adminKubeconfigSecret:
+    name: ci-cluster-v4-1-0-ndv7l-admin-kubeconfig
+  adminPasswordSecret:
+    name: ci-cluster-v4-1-0-ndv7l-admin-password
+  apiURL: https://api.ci-cluster-v4-1.r4f1.s1.devshift.org:6443
+  certificateBundles:
+  - generated: true
+    name: primary-cert-bundle
+  cliImage: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:528f2ead3d1605bdf818579976d97df5dd86df0a2a5d80df9aa8209c82333a86
+  clusterID: 1e4f00a3-a30e-46d7-9567-4bb61caf6d73
+  clusterVersionStatus:
+    [SNIP]
+  conditions:
+  - lastProbeTime: "2019-10-23T08:40:24Z"
+    lastTransitionTime: "2019-10-23T08:40:24Z"
+    message: Control plane certificates are present
+    reason: ControlPlaneCertificatesFound
+  installed: true
+  installedTimestamp: "2019-10-23T08:38:03Z"
+  installerImage: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:c9c1e6d1897b74d366d331f5cdc3fa16c5a96abe28ff8238833c09042156e983
+  provision:
+    name: ci-cluster-v4-1-0-ndv7l
+  webConsoleURL: https://console-openshift-console.apps.ci-cluster-v4-1.r4f1.s1.devshift.org
+```
+
+### Proposed Changes
+
+
+ 1. Add Spec.Installation struct to contain install specific, immutable settings.
+ 1. Add Spec.Installation.InstallConfigSecret containing a full openshift-install InstallConfig  to pass directly to installer.
+   * This will allow immediate use of new openshift-install features without needing to mirror in Hive's API.
+   * Ideally most items in ClusterDeployment.Spec should be assumed mutable, however this will not universally be true.
+   * There will be some overlap, information that must be provided in both the InstallConfig and the ClusterDeployment. We will attempt to error if you mismatch this information whenever possible.
```

Comment:
A "mismatch" may mean "change from installed value".  At creation time I would expect values to be the same but if it's mutable it can't be checked against the original install config on update.